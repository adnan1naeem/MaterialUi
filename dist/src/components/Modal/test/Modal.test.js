"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const enzyme_adapter_react_16_1 = __importDefault(require("enzyme-adapter-react-16"));
require("jest");
const __1 = require("../");
const Button_1 = __importDefault(require("@material-ui/core/Button"));
const enzyme_1 = require("enzyme");
enzyme_1.configure({ adapter: new enzyme_adapter_react_16_1.default() });
test('render correctly', () => {
    const notification = enzyme_1.shallow(react_1.default.createElement(__1.Modal, null,
        react_1.default.createElement(__1.ModalHeader, null, "UserLocation Service"),
        react_1.default.createElement(__1.ModalBody, null, "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running."),
        react_1.default.createElement(__1.ModalFooter, null,
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Disagree"),
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Agree"),
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Disagree"),
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Agree"))));
    expect(notification).toMatchSnapshot();
});
test('render correctly', () => {
    const notification = enzyme_1.shallow(react_1.default.createElement(__1.Modal, null,
        react_1.default.createElement(__1.ModalHeader, null, "UserLocation Service"),
        react_1.default.createElement(__1.ModalBody, null, "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running."),
        react_1.default.createElement(__1.ModalFooter, null,
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Disagree"),
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Agree"),
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Disagree"),
            react_1.default.createElement(Button_1.default, { onClick: () => {
                    ' ';
                }, color: "primary" }, "Agree"))));
    expect(notification).toMatchSnapshot();
});
test('Modal has prop fullScreen', () => {
    const notification = enzyme_1.shallow(react_1.default.createElement(__1.Modal, { fullScreen: true }));
    expect(notification.props().fullScreen).toEqual(true);
});
test('Modal has prop maxWidth', () => {
    const notification = enzyme_1.shallow(react_1.default.createElement(__1.Modal, { fullWidth: true, maxWidth: "xs" }));
    expect(notification.props().maxWidth).toEqual('xs');
});
test('Modal has prop fullScreen', () => {
    const notification = enzyme_1.shallow(react_1.default.createElement(__1.Modal, { fullScreen: true }));
    expect(notification.props().fullScreen).toEqual(true);
});
//# sourceMappingURL=Modal.test.js.map