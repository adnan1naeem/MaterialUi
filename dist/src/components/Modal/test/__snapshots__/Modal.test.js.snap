// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal>
    <ModalHeader>
        UserLocation Service
    </ModalHeader>
    <ModalBody>
        Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.
    </ModalBody>
    <ModalFooter>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Disagree
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Agree
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Disagree
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Agree
        </WithStyles(Button)>
    </ModalFooter>
</Modal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ModalHeader>
          UserLocation Service
</ModalHeader>,
        <ModalBody>
          Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.
</ModalBody>,
        <ModalFooter>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Disagree
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Agree
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Disagree
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Agree
          </WithStyles(Button)>
</ModalFooter>,
      ],
      "onClose": undefined,
      "onEscapeKeyDown": undefined,
      "open": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "UserLocation Service",
        },
        "ref": null,
        "rendered": "UserLocation Service",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
        },
        "ref": null,
        "rendered": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Disagree
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Agree
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Disagree
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Agree
</WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Disagree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Disagree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Agree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Agree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Disagree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Disagree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Agree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Agree",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ModalHeader>
            UserLocation Service
</ModalHeader>,
          <ModalBody>
            Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.
</ModalBody>,
          <ModalFooter>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Disagree
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Agree
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Disagree
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Agree
            </WithStyles(Button)>
</ModalFooter>,
        ],
        "onClose": undefined,
        "onEscapeKeyDown": undefined,
        "open": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "UserLocation Service",
          },
          "ref": null,
          "rendered": "UserLocation Service",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
          },
          "ref": null,
          "rendered": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Disagree
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Agree
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Disagree
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Agree
</WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Disagree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Disagree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Agree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Agree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Disagree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Disagree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Agree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Agree",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`render correctly 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal>
    <ModalHeader>
        UserLocation Service
    </ModalHeader>
    <ModalBody>
        Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.
    </ModalBody>
    <ModalFooter>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Disagree
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Agree
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Disagree
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            onClick={[Function]}
        >
            Agree
        </WithStyles(Button)>
    </ModalFooter>
</Modal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ModalHeader>
          UserLocation Service
</ModalHeader>,
        <ModalBody>
          Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.
</ModalBody>,
        <ModalFooter>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Disagree
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Agree
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Disagree
          </WithStyles(Button)>
          <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
          >
                    Agree
          </WithStyles(Button)>
</ModalFooter>,
      ],
      "onClose": undefined,
      "onEscapeKeyDown": undefined,
      "open": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "UserLocation Service",
        },
        "ref": null,
        "rendered": "UserLocation Service",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
        },
        "ref": null,
        "rendered": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Disagree
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Agree
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Disagree
</WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
>
              Agree
</WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Disagree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Disagree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Agree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Agree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Disagree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Disagree",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Agree",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Agree",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ModalHeader>
            UserLocation Service
</ModalHeader>,
          <ModalBody>
            Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.
</ModalBody>,
          <ModalFooter>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Disagree
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Agree
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Disagree
            </WithStyles(Button)>
            <WithStyles(Button)
                        color="primary"
                        onClick={[Function]}
            >
                        Agree
            </WithStyles(Button)>
</ModalFooter>,
        ],
        "onClose": undefined,
        "onEscapeKeyDown": undefined,
        "open": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "UserLocation Service",
          },
          "ref": null,
          "rendered": "UserLocation Service",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
          },
          "ref": null,
          "rendered": "Let Google help apps determine location. This means sending anonymous location data to Google, even when no apps are running.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Disagree
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Agree
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Disagree
</WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
>
                Agree
</WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Disagree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Disagree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Agree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Agree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Disagree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Disagree",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Agree",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Agree",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
